src/Endpoints/class-graphql-metadata.php

```diff
@@ -0,0 +1,266 @@
+		register_graphql_object_type( self::GRAPHQL_META_TYPE, $meta_type );
+
+		$container_type = array(
+			'description' => __( 'Parse.ly Metadata root type.', 'wp-parsely' ),
+			'fields'      => array(
+				'version'  => array(
+					'type'        => 'String',
+					'description' => __( 'Revision of the metadata format.', 'wp-parsely' ),
+				),
+				'meta'     => array(
+					'type'        => 'ParselyMeta',
+					'description' => __( 'Structured and filterable metadata.', 'wp-parsely' ),
+				),
+				'rendered' => array(
+					'type'        => 'String',
+					'description' => __( 'HTML string containing the metadata. Intended to be rendered in the front-end as is.', 'wp-parsely' ),
+				),
+			),
+		);
+		register_graphql_object_type( self::GRAPHQL_CONTAINER_TYPE, $container_type );
+	}
+
+	/**
+	 * Register the custom metadata fields so they can be queried in GraphQL.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	private function register_fields(): void {
+		$options      = $this->parsely->get_options();
+		$object_types = array_unique( array_merge( $options['track_post_types'], $options['track_page_types'] ) );
+
+		/**
+		 * Filters the list of post object types that the Parse.ly GraphQL API is hooked into.
+		 *
+		 * @since 3.2.0
+		 *
+		 * @param string[] $object_types Array of strings containing the object types, i.e. `page`, `post`, `term`.
+		 */
+		$object_types = apply_filters( 'wp_parsely_graphql_object_types', $object_types );
+
+		foreach ( $object_types as $object_type ) {
+			$post_type_object = get_post_type_object( $object_type );
+
+			register_graphql_field(
+				$post_type_object->graphql_single_name,
+				self::FIELD_NAME,
+				array(
+					'type'        => self::GRAPHQL_CONTAINER_TYPE,
+					'description' => 'Parse.ly metadata fields, to be rendered in the front-end so they can be parsed by the crawler. See https://www.parse.ly/help/integration/crawler.',
+					'resolve'     => function ( $graphql_post ) {
+						$post_id = $graphql_post->__get( 'ID' );
+						$post    = WP_Post::get_instance( $post_id );
+
+						if ( false === $post ) {
+							return array();
+						}
+
+						$meta = $this->parsely->construct_parsely_metadata( $this->parsely->get_options(), $post );
+						$meta = $this->process_meta_for_graphql( $meta );
+
+						/**
+						 * Filters the array with the actual metadata that is exposed through GraphQL.
+						 *
+						 * @see https://wpgraphqldocs.gatsbyjs.io/functions/register_graphql_field/
+						 * @since 3.2.0
+						 *
+						 * @param array $meta Array with the fields of the new type.
+						 */
+						$meta = apply_filters( 'wp_parsely_graphql_meta_object', $meta );
+
+						return array(
+							'version'  => self::GRAPHQL_VERSION,
+							'meta'     => $meta,
+							'rendered' => self::get_rendered_meta(),
+						);
+					},
+				)
+			);
+		}
+	}
+
+	/**
+	 * Adapts an array of metadata to GraphQL. Namely, all keys with a leading `@` symbol are duplicated by the same
```

Comment:
```suggestion
	 * Adapts an array of metadata to GraphQL. Namely, all keys with a leading `@` symbol are duplicated to the same
```

Also, this needs (in phpDoc Description, not Summary (first line)) an explanation as to why this is needed. Maybe GraphQL doesn't support `@` in keys? Maybe something else?

Reply:
> Maybe GraphQL doesn't support @ in keys

Yes, because `@` is reserved for [directives](http://graphql.org/learn/queries/#directives).

Reply:
TIL - I suspected that would be the case, but referencing the avoidance of clashing with directives would be good rationale to include in the docs here IMO.